### PR TITLE
Core: fix PnP compatibility for @storybook/ui and @storybook/router packages

### PR DESCRIPTION
**Pull request by Andarist**
Issue: https://github.com/storybookjs/storybook/issues/18319

It's pretty much the same as https://github.com/storybookjs/storybook/pull/18015/files

### CODE REVIEW

File: lib/ui/package.json [hunk 1]
```diff
@@ -50,6 +50,7 @@
     "@storybook/semver": "^7.3.2",
     "@storybook/theming": "6.5.0-rc.1",
     "core-js": "^3.8.2",
+    "qs": "^6.10.0",
```
Comment (Andarist) [inline lib/ui/package.json:53]: The reported issue was only related to this package

File: lib/router/package.json [hunk 2]
```diff
@@ -42,6 +42,7 @@
   "dependencies": {
     "@storybook/client-logger": "6.5.0-rc.1",
     "core-js": "^3.8.2",
+    "qs": "^6.10.0",
```
Comment (Andarist) [inline lib/router/package.json:45]: but I've audited the whole `lib` direcotry and found this offender too

**Comment by yannbf** [comment]
That's amazing @Andarist ! I'm updating the branch to reenable the E2E tests that were failing because of this issue, I hope you don't mind!

**Comment by yannbf** [comment]
@Andarist after reenabling the e2e tests, seems like it still fails, however not with the same issue (which is great!)

https://app.circleci.com/pipelines/github/storybookjs/storybook/26404/workflows/a6f23f92-f687-4ae2-8f19-71a8c39964fc/jobs/385085

**Comment by shilman** [comment]
@Andarist it looks like the PnP tests are failing for CRA. Is this something you can look into? If you'd like to pair I'd be happy to look at it together. I'm not sure whether it's related to PnP or just a problem with CRA in the current state of things.

**Comment by Andarist** [comment]
@yannbf @shilman I think that you are referring to the same problem - correct me if I'm wrong.

So I can easily reproduce this by just calling `require('@storybook/react')` in node. This is because its main field points to a file within `/client/` directory:
https://github.com/storybookjs/storybook/blob/6b8210bb0d760e0bdda4be801f167d9db42cc50b/app/react/package.json#L22
this in turn leads to executing this call at the initialization time:
https://github.com/storybookjs/storybook/blob/6b8210bb0d760e0bdda4be801f167d9db42cc50b/app/react/src/client/preview/index.tsx#L20
and that leads to creating a `PostmsgTransport` and calling this in its constructor:
https://github.com/storybookjs/storybook/blob/6b8210bb0d760e0bdda4be801f167d9db42cc50b/lib/channel-postmessage/src/index.ts#L38

The problem is that `globalWindow` is not really a window in node, it's a process object, so it doesn't have an `addEventListener` method. Note that `globalWindow` actually is being (sort of) retrieved through an external package but that doesn't change a lot because its main file is super simple:
https://github.com/Raynos/global/blob/91c4362aa2720f7c7a3a085707de7e04f5e77cf3/window.js

Question would be - should node ever execute what is inside that `/client/` directory? If not then why did it? Can you provide me with a full script that repro this in a more full project? I've tried to follow your CI setup but it isn't super straightforward and I didn't have enough time right now to analyze it further. 

Either way - this doesn't look related to this particular PR and to PnP.

**Comment by yannbf** [comment]
> 

Great point @Andarist! I think I know what's going on and I'll attempt a small change in this PR to check whether the problem will be fixed.

**Comment by Andarist** [comment]
@yannbf it looks like the e2e and regressions tests are ✅

**Review by ndelangen — APPROVED** [review]

**Comment by yannbf** [comment]
Thanks a lot for your work @Andarist <3

**Pull request merged**